kolibri/core/auth/management/commands/sync.py

```diff
@@ -32,52 +37,73 @@ def add_arguments(self, parser):
             default=500,
             help="Chunk size of records to send/retrieve per request",
         )
-
-    def _fullfacilitysync(self, baseurl, facility=None, chunk_size=500):
-        if facility is None:
-            dataset_id = None
-        else:
-            dataset_id = facility.dataset_id
-
-        if baseurl != DATA_PORTAL_SYNCING_BASE_URL:
-            self.stdout.write("Syncing has been initiated (this may take a while)...")
-            call_command(
-                "fullfacilitysync",
-                base_url=baseurl,
-                dataset_id=dataset_id,
-                chunk_size=chunk_size,
-            )
-            self.stdout.write("Syncing has been completed.")
-            sys.exit(0)
+        parser.add_argument(
+            "--no-push", action="store_true", help="Do not push data to the server"
+        )
+        parser.add_argument(
+            "--no-pull", action="store_true", help="Do not pull data from the server"
+        )
+        parser.add_argument(
+            "--username",
+            type=str,
+            help="username of superuser on server we are syncing with",
+        )
+        parser.add_argument(
+            "--password",
+            type=str,
+            help="password of superuser on server we are syncing with",
+        )
+        # parser.add_argument("--scope-id", type=str, default=FULL_FACILITY)
 
     def handle_async(self, *args, **options):
 
-        baseurl = options["baseurl"]
-        facility_id = options["facility"]
-        chunk_size = options["chunk_size"]
+        baseurl, facility_id, chunk_size, username, password, no_push, no_pull, noninteractive = (
+            options["baseurl"],
+            options["facility"],
+            options["chunk_size"],
+            options["username"],
+            options["password"],
+            options["no_push"],
+            options["no_pull"],
+            options["noninteractive"],
+        )
 
-        # This handles the case for when we want to pull in facility data for our empty kolibri instance
-        if not facility_id:
-            self._fullfacilitysync(baseurl)
+        PORTAL_SYNC = baseurl == DATA_PORTAL_SYNCING_BASE_URL
 
-        facility = get_facility(
-            facility_id=facility_id, noninteractive=options["noninteractive"]
-        )
+        # validate url that is passed in
+        if not PORTAL_SYNC:
+            try:
+                baseurl = NetworkClient(address=baseurl).base_url
+            except URLParseError:
+                raise CommandError(
+                    "Base URL/IP: {} is not valid. Please retry command and enter a valid URL/IP.".format(
+                        baseurl
+                    )
+                )
```

Comment:
For later: what if it's valid, but it can't connect?

Reply:
Can add that exception and a nice error message